tracks/java/exercises/word-count/mentoring.md

```diff
@@ -0,0 +1,51 @@
+### Reasonable solutions
+
+Loop-based solution:
+```java
+import java.util.HashMap;
+import java.util.Map;
+
+class WordCount {
+
+    public Map<String, Integer> phrase(String input) {
+        Map<String, Integer> countMap = new HashMap<>();
+        input = input.trim().toLowerCase().replaceAll("([^A-Za-z0-9']|\\B'|'\\B)", " ");
+        String[] tokenizedInput = input.split("\\s+");
+        for (String aWord : tokenizedInput) {
+            if (aWord.isEmpty()) {
+                continue;
+            }
+            Integer count = countMap.get(aWord);
+            countMap.put(aWord, count == null ? 1 : count + 1);
+        }
+        return countMap;
+    }
+}
+```
+
+FP solution (Java 11):
```

Comment:
Consider writing out FP as 
```suggestion
Functional programming (FP) solution (Java 11+):
```

Reply:
I actually think most (if not all) of this actually works starting in Java 8, but I could be mistaken.

Reply:
@jmrunkle, I agree that **most** of this working in Java 8. But `java.util.function.Predicate.not;` is Java 11... So what should I do? ;-)

Reply:
Perhaps you can add that as a helper with the comment "Use java.util.function.Predicate.not in Java 11+"

Reply:
I guess I had forgotten that they didn't add that until Java 11. My company uses Java 8, but uses the Guava method so I just did not really notice: https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Predicates.html#not-com.google.common.base.Predicate-